### PR TITLE
[feat] Fuse batchNorm layer and enable reparameterization

### PR DESCRIPTION
**Pull request by hglee98**
## Description

This PR aims to present the reparameterization feature proposed in [RepVGG: Making VGG-style ConvNets Great Again](https://arxiv.org/abs/2101.03697)

Related to #529 

We recommend to link at least one existing issue for PR. Before your create a PR, please check if there is an issue for this change.  

## Change(s)

- Define abstracted deploy method to `TaskModel`
- Define deploy methods to class of each taskmodel.
- Convert model into deploy mode at the end of the training.

### results
before reparam
Screenshot 2024-09-05 at 2 16 50 PM

after reparam
Screenshot 2024-09-05 at 2 16 03 PM

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`. 

## Changelog

If your PR is granted to `dev` branch, codeowner will add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file including a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

Please enable **Allow edits and access to secrets by maintainers** so that our maintainers can update the `CHANGELOG.md`.

### CODE REVIEW

File: src/netspresso_trainer/models/base.py [hunk 1]
```diff
@@ -67,6 +67,10 @@ def _get_name(self):
     def forward(self, x, label_size=None, targets=None):
         raise NotImplementedError
 
+    @abstractmethod
+    def deploy(self, ):
+        raise NotImplementedError
```
Comment (illian01) [inline src/netspresso_trainer/models/base.py:72]: I think better to implement deploy method in `TaskModel` class instead of subclasses.

Comment (hglee98) [reply src/netspresso_trainer/models/base.py:72]: Yeah! Thank you for the review. I agree👍 It would prevent duplicates on the code base.

Comment (hglee98) [reply src/netspresso_trainer/models/base.py]: I made a commit to resolve this conversation!

**Review by illian01 — APPROVED** [review]
LGTM!

**Pull request merged**